MediaBrowser.Common/Plugins/BasePlugin.cs

```diff
@@ -245,15 +245,12 @@ public void SetStartupInfo(Action<string> directoryCreateFn)
         private TConfigurationType LoadConfiguration()
         {
             var path = ConfigurationFilePath;
-
-            try
-            {
-                return (TConfigurationType)XmlSerializer.DeserializeFromFile(typeof(TConfigurationType), path);
-            }
-            catch
+            if (!File.Exists(path))
             {
-                return (TConfigurationType)Activator.CreateInstance(typeof(TConfigurationType));
+                return Activator.CreateInstance<TConfigurationType>();
             }
+
+            return (TConfigurationType)XmlSerializer.DeserializeFromFile(typeof(TConfigurationType), path);
```

Comment:
Should still catch exceptions, we don't want the `LoadConfiguration` to cause startup to fail

Reply:
What kinds of exception might cause this to fail? Wrong xml format? I think hidding the dirt under the rug is not a good idea. We should probably catch only on startup and log it if something goes wrong and fix it later.

Reply:
Incorrect XML, file deleted, wrong permissions, the list goes on

Reply:
I agree that we can have several kinds of exceptions, but we should handle exception by their type not by any "Exception". On the file deleted exception i have checked it exits. If it doesn't i return a new instance of the default configuration.

Reply:
> I agree that we can have several kinds of exceptions, but we should handle exception by their type not by any "Exception".

Ofc

> On the file deleted exception i have checked it exits.

Technically it could get deleted between the check and you opening it.

Reply:
Technically yes, like almost any kind of operation... most operations are not atomic...